### PR TITLE
d/glue_catalog_table

### PR DESCRIPTION
**Pull request by chris922**
This is my first contribution, hope I did everything as expected. If not please let me know :) 

I added a data source for a glue catalog table and more or less copied and adapted many things from the regular resource. I was unsure if it is desired to have one generic "read" function and use this one in the resource and data source, but for simplicity, I now started with separate functions as for other resources it looks similar.

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccGlueDataCatalogDataSource*' ACCTEST_PARALLELISM=5 PKG=glue
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/glue/... -v -count 1 -parallel 5  -run=TestAccGlueDataCatalogDataSource* -timeout 180m
=== RUN   TestAccGlueDataCatalogDataSource_basic
=== PAUSE TestAccGlueDataCatalogDataSource_basic
=== CONT  TestAccGlueDataCatalogDataSource_basic
--- PASS: TestAccGlueDataCatalogDataSource_basic (29.75s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/glue       29.790s
...
```

### CODE REVIEW

**Comment by bschaatsbergen** [comment]
Hi @chris922, thanks for raising this pull request. I've noted it down to review this in the coming days.

File: website/docs/d/glue_catalog_table.html.markdown [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+* `bucket_columns` - List of reducer grouping columns, clustering columns, and bucketing columns in the table.
+* `columns` - Configuration block for columns in the table. See [`columns`](#columns) below.
+* `compressed` - Whether the data in the table is compressed.
+* `input_format` - Input format: SequenceFileInputFormat (binary), or TextInputFormat, or a custom format.
+* `location` - Physical location of the table. By default, this takes the form of the warehouse location, followed by the database location in the warehouse, followed by the table name.
+* `number_of_buckets` - Is if the table contains any dimension columns.
+* `output_format` - Output format: SequenceFileOutputFormat (binary), or IgnoreKeyTextOutputFormat, or a custom format.
+* `parameters` - User-supplied properties in key-value form.
+* `schema_reference` - Object that references a schema stored in the AWS Glue Schema Registry. See [Schema Reference](#schema-reference) below.
+* `ser_de_info` - Configuration block for serialization and deserialization ("SerDe") information. See [`ser_de_info`](#ser_de_info) below.
```
Comment (bschaatsbergen) [inline website/docs/d/glue_catalog_table.html.markdown:71]: Lets create some consistency between the paragraph references, I noticed diferent ways being used.

>  See [`ser_de_info`](#ser_de_info) below
> See [Schema Reference](#schema-reference) below

**Review by bschaatsbergen — CHANGES_REQUESTED** [review]
Thanks for your contribution @chris922 , overall it looks good to m. I've used the data source and ran the tests on your fork - which seemed 👍 . 

```
bruno@bruno ~/g/f/chris992-terraform-provider-aws (feature/glue-data-catalog-table-datasource)> make testacc TESTARGS='-run=TestAccGlueDataCatalogDataSource*' ACCTEST_PARALLELISM=5 PKG=glue
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/glue/... -v -count 1 -parallel 5  -run=TestAccGlueDataCatalogDataSource* -timeout 180m
=== RUN   TestAccGlueDataCatalogDataSource_basic
=== PAUSE TestAccGlueDataCatalogDataSource_basic
=== CONT  TestAccGlueDataCatalogDataSource_basic
--- PASS: TestAccGlueDataCatalogDataSource_basic (26.60s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/glue       28.114s
```

If you could address the few comments we can get this 🚢  shipped!

**Comment by chris922** [comment]
Thanks @bschaatsbergen for the review! Please have a look at my changes and resolve the threads if they match your expectation.

**Comment by chris922** [comment]
Hmm, I squashed and rebased it... but the pipeline still fails, any ideas?

**Comment by bschaatsbergen** [comment]
I'll take a look at this tomorrow.

**Comment by chris922** [comment]
@bschaatsbergen : I rebased my changes onto the main and it looks better for now

**Comment by bschaatsbergen** [comment]
@ewbankkit would you mind checking this PR out :) ?

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccGlueCatalogTableDataSource_' PKG=glue ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/glue/... -v -count 1 -parallel 2  -run=TestAccGlueCatalogTableDataSource_ -timeout 180m
=== RUN   TestAccGlueCatalogTableDataSource_basic
=== PAUSE TestAccGlueCatalogTableDataSource_basic
=== CONT  TestAccGlueCatalogTableDataSource_basic
--- PASS: TestAccGlueCatalogTableDataSource_basic (17.94s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/glue	23.130s
```

**Comment by chris922** [comment]
Great, thanks a lot!

**Pull request merged**